pandas/core/frame.py

```diff
@@ -3995,20 +3995,37 @@ def rename(self, *args, **kwargs):
         We *highly* recommend using keyword arguments to clarify your
         intent.
 
+        Rename columns using a mapping:
         >>> df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
-        >>> df.rename(index=str, columns={"A": "a", "B": "c"})
+        >>> df.rename(columns={"A": "a", "B": "c"})
            a  c
         0  1  4
         1  2  5
         2  3  6
 
-        >>> df.rename(index=str, columns={"A": "a", "C": "c"})
+        Rename rows using a mapping:
+        >>> df = pd.DataFrame(data={"A": [1, 2, 3], "B": [4, 5, 6]},
+        ...                   index=["X", "Y", "Z"])
+        >>> df.rename(index={"X": "x", "Y": "y", "Z": "z"})
+           A  B
+        x  1  4
+        y  2  5
+        z  3  6
+
+        Cast index labels to a different type:
+        >>> df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
```

Comment:
Once the above example is removed, you don't need to construct the `df` twice in this example

Reply:
example above won't be removed

Reply:
I see what you mean now. Example is removed.